src/packages/recompose/isReferentiallyTransparentFunctionComponent.js

```diff
@@ -5,7 +5,7 @@ const isReferentiallyTransparentFunctionComponent = Component => Boolean(
   !isClassComponent(Component) &&
   !Component.defaultProps &&
   !Component.contextTypes &&
-  !Component.propTypes
+  (!Component.propTypes || process.env.NODE_ENV === 'production')
```

Comment:
@istarkov I agree that it would be better. I believe that uglify-js is smart enough to remove those `if (!foo.bar || true)` expression but that may not be true with other minifier, they may keep it.